gesturelibrary/src/gesturelib.c

```diff
@@ -10,19 +10,26 @@ gesture_recognizer_t recognizers[MAX_RECOGNIZERS];
 int num_recognizers = 0;
 
 /// @brief set containing most recent touch within finger group
-touch_event_t* groups_heads[MAX_TOUCHES];
+touch_event_t latest_touch_events[MAX_TOUCHES];
 touch_event_t empty_touch_event = {
     .event_type = TOUCH_EVENT_UP, .position_x = 0, .position_y = 0, .timestamp = 0, .id = TOUCH_ID_UNDEFINED};
 
-static float squared_distance(touch_event_t* a, touch_event_t* b);
+static float squared_distance(touch_event_t a, touch_event_t b);
+
+/// @brief assign a group ID to this touch_event. Update group_heads to reflect this. If this is not a DOWN event
+///        and there are no groups being tracked, do not assign an ID. If this is a DOWN event but MAX_TOUCHES groups
+///        are already being tracked, do not assign an ID.
+/// @param touch_event
+/// @return
+static unsigned int assign_group(touch_event_t* touch_event);
 
 /** This is a documentation test. **/
 void init_gesturelib() {
     for (uint32_t index = 0; index < MAX_RECOGNIZERS; index++) {
         recognizers[index].enabled = 0;
     }
     for (unsigned int i = 0; i < MAX_TOUCHES; i++) {
-        groups_heads[i] = &empty_touch_event;
+        latest_touch_events[i] = empty_touch_event;
```

Comment:
please assign these after calling all recognizers
this way recognizers are able to compare the current with the previous touch events

Reply:
I did. This is just in the init function. Refer to the process function and the description on the PR